lib/readthis/cache.rb

```diff
@@ -352,6 +355,10 @@ def extract_options!(array)
     end
 
     def merged_options(options)
```

Comment:
Let's move the casting to integer just into the `merged_options` method, and clean up the checking a little bit. Anything that was passed as an `expires_in` option must respond to `to_i`, including any number, so it will be called every time `merged_options` is anyhow. Not to mention the result of calling `1.to_i` is faster than `1.to_i if 1.respond_to?(:to_i)`:

```ruby
def merged_options(options)
  merged = @options.merge(options || {})

  if expires_in = merged[:expires_in]
    merged[:expires_in] = expires_in.to_i
  end

  merged
end
```

This also has options reversal fix that was in #28, which would have needed a rebase anyhow.